# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1425593123 -3600
#      Thu Mar 05 23:05:23 2015 +0100
# Node ID 7d888f9e94eb5fc32e27ec34314fd25779d4a9ac
# Parent  f8fbb7b9d78e7937dc10c0908ad052c37f27cc90
fix some shortcut issues

* QTerminal.h (constructor): make sure interrupt action signal is connected

* main-window.cc (notice_settings): set all shortcuts respecting the
  actual focus

diff --git a/libgui/qterminal/libqterminal/QTerminal.h b/libgui/qterminal/libqterminal/QTerminal.h
--- a/libgui/qterminal/libqterminal/QTerminal.h
+++ b/libgui/qterminal/libqterminal/QTerminal.h
@@ -168,17 +168,17 @@ protected:
 
     // extra interrupt action
     _interrupt_action = new QAction (this);
     addAction (_interrupt_action);
 
     _interrupt_action->setShortcut (
             QKeySequence (Qt::ControlModifier + Qt::Key_C));
 
-    connect (_interrupt_action, SIGNAL (triggered (bool)),
+    connect (_interrupt_action, SIGNAL (triggered ()),
             this, SLOT (terminal_interrupt ()));
   }
 
 private:
 
   QMenu *_contextMenu;
   QAction * _copy_action;
   QAction * _paste_action;
diff --git a/libgui/src/main-window.cc b/libgui/src/main-window.cc
--- a/libgui/src/main-window.cc
+++ b/libgui/src/main-window.cc
@@ -742,24 +742,27 @@ main_window::notice_settings (const QSet
 
   if (settings->value ("show_status_bar",true).toBool ())
     status_bar->show ();
   else
     status_bar->hide ();
 
   _prevent_readline_conflicts =
     settings->value ("shortcuts/prevent_readline_conflicts", true).toBool ();
-  set_global_shortcuts (command_window_has_focus ());
 
   _suppress_dbg_location =
         ! settings->value ("terminal/print_debug_location", false).toBool ();
 
   resource_manager::update_network_settings ();
 
   emit active_dock_changed (0, _active_dock); // update dock widget styles
+
+  configure_shortcuts ();
+  set_global_shortcuts (_active_dock == command_window);
+  set_global_edit_shortcuts (_active_dock == editor_window);
 }
 
 void
 main_window::confirm_shutdown_octave (void)
 {
   bool closenow = true;
 
   QSettings *settings = resource_manager::get_settings ();
